app/CommandHandling/ContextFactory.php

```diff
@@ -80,4 +69,15 @@ public static function prepareForLogging(Metadata $metadata): Metadata
 
         return new Metadata($metadata);
     }
+
+    public static function createFromGlobals(ContainerInterface $container): Metadata
```

Comment:
My 2 cents: I wouldn't pass the container to classes but inject the properties directly. Unless there's a reason that can't be done?

Reply:
In general I agree, but in some cases we do this because some things like `CurrentUser` are not always set immediately but only when a Request middleware is executed. So we sometimes need to delay the fetching until the last moment. I'm not 100% sure if it's needed here, but I figured it would be best to leave it like it is (but refactor it to use `ContainerInterface` instead of `Application`), and then revisit it later to figure out the details

Reply:
(Not sure if it was clear but the `createFromGlobals()` already existed, Jonas removed it and injected the properties directly into the constructor to get rid of the `Application` dependency, but I re-added it to be safe and refactored it to use `ContainerInterface` instead)